frontend-public/src/components/ReturnPage.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import { getRedirect } from "@/selectors/authenticationSelectors";
+import {
+  authenticateUser,
+  unAuthenticateUser,
+  signOutFromSSO,
+} from "@/actionCreators/authenticationActionCreator";
+import queryString from "query-string";
+import { RETURN_PAGE_TYPE } from "../constants/strings";
+import Loading from "@/components/common/Loading";
+
+export class ReturnPage extends Component {
+  static propTypes = {
+    location: PropTypes.shape({ search: PropTypes.string }).isRequired,
+    unAuthenticateUser: PropTypes.func.isRequired,
+    authenticateUser: PropTypes.func.isRequired,
+    redirect: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
+  };
+
+  componentDidMount() {
+    // grab the code and redirect type from the redirect url
+    const { type, code } = queryString.parse(this.props.location.search);
+    switch (type) {
+      case RETURN_PAGE_TYPE.LOGIN:
+        if (code) {
```

Comment:
Does this need to take any action if the code isn't present?

Reply:
simplified.